src/Catalyst.Core.Modules.Rpc.Server/RpcServer.cs

```diff
@@ -0,0 +1,84 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.IO.EventLoop;
+using Catalyst.Abstractions.IO.Messaging.Dto;
+using Catalyst.Abstractions.IO.Observers;
+using Catalyst.Abstractions.IO.Transport.Channels;
+using Catalyst.Abstractions.Rpc;
+using Catalyst.Core.Lib.IO.Transport;
+using Catalyst.Protocol.Wire;
+using Serilog;
+
+namespace Catalyst.Core.Modules.Rpc.Server
+{
+    public sealed class RpcServer : TcpServer, IRpcServer
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IDisposable' in this class and use the 'Dispose' method to call 'Dispose' on '_cancellationSource'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)